tests/base/converters/test_dataframe_to_mds.py

```diff
@@ -0,0 +1,322 @@
+                    shards = json.load(open(os.path.join(sub_dir, 'index.json'), 'r'))['shards']
+                    if shards:
+                        nsamples += shards[0]['samples']
+            assert (nsamples == sum([a['samples'] for a in mgi['shards']]))
+        else:
+            assert not (os.path.exists(os.path.join(
+                out, 'index.json'))), 'merged index is created when merge_index=False'
+
+    @pytest.mark.parametrize('use_columns', [True, False])
+    def test_end_to_end_conversion_local_decimal(self, decimal_dataframe: Any, use_columns: bool):
+        out = mkdtemp()
+        user_defined_columns = {'id': 'int', 'name': 'str', 'amount': 'str_decimal'}
+        mds_kwargs = {
+            'out': out,
+            'columns': user_defined_columns,
+            'keep_local': True,
+            'compression': 'zstd:7',
+            'hashes': ['sha1', 'xxh64'],
+            'size_limit': 1 << 26
+        }
+
+        if use_columns:
+            mds_kwargs['columns'] = user_defined_columns
+
+        _, _ = dataframeToMDS(decimal_dataframe, merge_index=True, mds_kwargs=mds_kwargs)
+        assert (len(os.listdir(out)) > 0), f'{out} is empty'
+
+    @pytest.mark.parametrize('user_defined_columns', [{
+        'idd': 'str',
+        'dept': 'str'
+    }, {
+        'id': 'strr',
+        'dept': 'str'
+    }])
+    def test_user_defined_columns(self, dataframe: Any, user_defined_columns: Dict):
+        out = mkdtemp()
+        mds_kwargs = {
+            'out': out,
+            'columns': user_defined_columns,
+            'keep_local': True,
+            'compression': 'zstd:7',
+            'hashes': ['sha1', 'xxh64'],
+            'size_limit': 1 << 26
+        }
+        with pytest.raises(ValueError):
+            _, _ = dataframeToMDS(dataframe, merge_index=False, mds_kwargs=mds_kwargs)
+
+    @pytest.mark.parametrize('keep_local', [True, False])
+    @pytest.mark.parametrize('merge_index', [True, False])
+    def test_end_to_end_conversion_local(self, dataframe: Any, keep_local: bool,
+                                         merge_index: bool):
+        out = mkdtemp()
+        mds_kwargs = {
+            'out': out,
+            'columns': {
+                'id': 'str',
+                'dept': 'str'
+            },
+            'keep_local': keep_local,
+            'compression': 'zstd:7',
+            'hashes': ['sha1', 'xxh64'],
+            'size_limit': 1 << 26
+        }
+
+        _, _ = dataframeToMDS(dataframe, merge_index=merge_index, mds_kwargs=mds_kwargs)
+
+        assert (len(os.listdir(out)) > 0), f'{out} is empty'
+        for d in os.listdir(out):
+            if os.path.isdir(os.path.join(out, d)):
+                assert (os.path.exists(os.path.join(
+                    out, d, 'index.json'))), f'No index.json found in subdirectory {d}'
+
+        if merge_index == True:
+            assert (os.path.exists(os.path.join(out, 'index.json'))), 'No merged index found'
```

Comment:
`No merged index.json found`. Check the similar error message to other places as well.